src/cephadm/cephadmlib/daemons/__init__.py

```diff
@@ -0,0 +1,3 @@
+from .custom import CustomContainer
+
+__all__ = ['CustomContainer']
```

Comment:
Haven't looked much at `__all__` before so trying to get a full understanding of the purpose here. What I found was that this affects `from <mod> import *` so that it only import things listed in `__all__`, rather than just every symbol without an underscore which seems to be the default. Is this just for tracking what we have or do we actually expect to have things we don't want to be imported by a `from cephadmlib.daemons import *`

Reply:
That's *a* reason. The primary reason is that the linter complains about unused imports if you only import the classes. However, the classes (or really the modules containing them) *need* to be imported in order to call the class decorator that registers the daemon types.

This way code that imports `cephadmlib.daemons` will be able to use all the daemon forms functionality that uses the registered daemon form classes & daemon type names without angering the linter *and* we have a clean constrained list of what you can import out of `cephadmlib.daemons` directly (you can always still import `cephadm.daemons.foo` as needed).

Reply:
Ah, okay. Hadn't thought about the linter at all, or realized that constraint of registering the daemon types when moving them out of cephadm.py . That makes sense.